File: control-assurance-sp/src/types/GIAAUpdate.ts
import { GIAAUpdateTypes } from "./AppGlobals";
import { IEntity } from "./Entity";


export interface IGIAAUpdate  extends IEntity {

    UpdateType?: string;
    UpdateDetails?: string;
    RevisedDate?: Date;
    Link?: string;
    EvFileName?: string;
    EvIsLink?:boolean;
    EvAdditionalNotes?: string;
    GIAARecommendationId?: number;
    GIAAActionStatusTypeId?: number;
    UpdatedById?: number;
    UpdateDate?: Date;
    UpdateChangeLog?: string;
    RequestClose?:boolean;
    RequestDateChange?:boolean;
    RequestDateChangeTo?: Date;
    MarkAllReqClosed?:boolean;
    

}

export class GIAAUpdate implements IGIAAUpdate{ 
    public ID: number = 0;
    public Title: string = null;    

    public UpdateType?: string = null;
    public UpdateDetails?: string = null;
    public RevisedDate?: Date = null;
    public Link?: string = null;
    public EvFileName?: string = null;
    public EvIsLink?:boolean = false;
    public EvAdditionalNotes?: string = null;
    public GIAARecommendationId?: number = null;
    public GIAAActionStatusTypeId?: number = null;
    public UpdatedById?: number = null;
    public UpdateDate?: Date = null;
    public UpdateChangeLog?: string = null;
    public RequestClose?:boolean = false;
    public RequestDateChange?:boolean = false;
    public RequestDateChangeTo?: Date = null;
    public MarkAllReqClosed?:boolean = null;

    constructor(giaaRecommendationId: number, updateType: string) {
        this.GIAARecommendationId = giaaRecommendationId;
        this.UpdateType = updateType;

        if(updateType === GIAAUpdateTypes.Status_DateUpdate){
            this.MarkAllReqClosed = true; //default checked
        }
    }


}

File: ControlAssuranceAPI/Models/GIAAUpdate.cs
using System;
using System.Collections.Generic;

namespace CAT.Models;

public partial class GIAAUpdate
{
    public int ID { get; set; }

    public string? Title { get; set; }

    public string? UpdateType { get; set; }

    public string? UpdateDetails { get; set; }

    public DateTime? RevisedDate { get; set; }

    public string? Link { get; set; }

    public string? EvFileName { get; set; }

    public bool? EvIsLink { get; set; }

    public string? EvAdditionalNotes { get; set; }

    public int? GIAARecommendationId { get; set; }

    public int? GIAAActionStatusTypeId { get; set; }

    public int? UpdatedById { get; set; }

    public DateTime? UpdateDate { get; set; }

    public string? UpdateChangeLog { get; set; }

    public bool? RequestClose { get; set; }

    public bool? RequestDateChange { get; set; }

    public DateTime? RequestDateChangeTo { get; set; }

    public bool? RequestStatusOpen { get; set; }

    public bool? MarkAllReqClosed { get; set; }

    public virtual GIAAActionStatusType? GIAAActionStatusType { get; set; }

    public virtual GIAARecommendation? GIAARecommendation { get; set; }

    public virtual User? UpdatedBy { get; set; }
}

public static class GIAAUpdateTypes
{
    public readonly static string ActionUpdate = "Action Update";
    public readonly static string Status_DateUpdate = "Status/Date Update";
    public readonly static string GIAAComment = "GIAA Comment";
    public readonly static string MiscComment = "Misc Comment";
    public readonly static string RecChanged = "Rec Changed";
    public readonly static string GIAAUpdate = "GIAA Update";
}
public class GIAAUpdateView_Result
{
    public int ID { get; set; }
    public string? Title { get; set; }
    public string? UpdateType { get; set; }
    public string? UpdateBy { get; set; }
    public string? UpdateDate { get; set; }
    public string? UpdateDetails { get; set; }
    public string? Requests { get; set; }
    public string? Status { get; set; }
    public string? RevisedDate { get; set; }
    public string? Evidence { get; set; }
    public bool EvIsLink { get; set; }
    public string? EvType { get; set; }
}